libcudacxx/include/cuda/std/detail/libcxx/include/span

```diff
@@ -585,6 +594,13 @@ public:
       , __size_{0}
   {}
 
+  _CCCL_TEMPLATE(class _Tp2 = _Tp)
+  _CCCL_REQUIRES(_CCCL_TRAIT(is_const, _Tp2))
+  _LIBCUDACXX_HIDE_FROM_ABI constexpr span(_CUDA_VSTD::initializer_list<value_type> __il) noexcept
```

Comment:
This needs to remain explicit, its the test that are actually invalid and should explicitly create the constarray rather than relying on `{{1, 2}}`